external-crates/move/crates/move-analyzer/src/inlay_hints.rs

```diff
@@ -25,43 +25,45 @@ pub fn on_inlay_hint_request(context: &Context, request: &Request, symbols: &Sym
     );
     let mut hints: Vec<InlayHint> = vec![];
 
-    if context.inlay_type_hints {
-        if let Some(file_defs) = symbols.file_mods.get(&fpath) {
-            for mod_defs in file_defs {
-                for untyped_def_loc in mod_defs.untyped_defs() {
-                    let start_position = symbols.files.start_position(untyped_def_loc);
-                    if let Some(DefInfo::Local(n, t, _, _)) = symbols.def_info(untyped_def_loc) {
-                        let position = Position {
-                            line: start_position.line_offset() as u32,
-                            character: start_position.column_offset() as u32 + n.len() as u32,
-                        };
-                        let colon_label = InlayHintLabelPart {
-                            value: ": ".to_string(),
-                            tooltip: None,
-                            location: None,
-                            command: None,
-                        };
-                        let type_label = InlayHintLabelPart {
-                            value: type_to_ide_string(t, /* verbose */ true),
-                            tooltip: None,
-                            location: None,
-                            command: None,
-                        };
-                        let h = InlayHint {
-                            position,
-                            label: InlayHintLabel::LabelParts(vec![colon_label, type_label]),
-                            kind: Some(InlayHintKind::TYPE),
-                            text_edits: None,
-                            tooltip: additional_hint_info(t, symbols),
-                            padding_left: None,
-                            padding_right: None,
-                            data: None,
-                        };
-                        hints.push(h);
+    if let Some(symbols) =
+        SymbolicatorRunner::root_dir(&fpath).and_then(|pkg_path| symbols_map.get(&pkg_path))
+    {
+        if context.inlay_type_hints {
+            if let Some(file_defs) = symbols.file_mods.get(&fpath) {
```

Comment:
It would be a lot cleaner to pull this out into a function that returns `Option<Vec<InlayHint>>` so you could do these as serial `let` statements with `?` at the end of each.

Reply:
Good idea!